### PR TITLE
feat: add body text component

### PR DESCRIPTION
**Pull request by briantstephan**
Tested with a local VE build in dev mode and the component worked as expected.

### CODE REVIEW

**Comment by jwartofsky-yext** [comment]
Were you able to test this?

File: src/components/puck/BodyText.tsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      { label: "Normal", value: "normal" },
+      { label: "Medium", value: "medium" },
+      { label: "Semibold", value: "semibold" },
+      { label: "Bold", value: "bold" },
+      { label: "Extrabold", value: "extrabold" },
+      { label: "Black", value: "black" },
+    ],
+  },
+  color: {
+    label: "Text color",
```
Comment (asanehisa) [inline src/components/puck/BodyText.tsx:59]: just Color

File: src/components/puck/BodyText.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  label: "Body Text",
+  fields: bodyTextFields,
+  defaultProps: {
+    text: {
+      field: "",
+      constantValue: "Text",
+      constantValueEnabled: true,
+    },
+    fontWeight: "normal",
+    color: "text",
```
Comment (asanehisa) [inline src/components/puck/BodyText.tsx:92]: i think we want all the default props to use theme stuff, so 

fontWeight: "default" aka ""
color: "default"

File: src/components/puck/atoms/body.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import * as React from "react";
+import { cva, type VariantProps } from "class-variance-authority";
+import { cn } from "../../../internal/utils/cn.ts";
+
+// Define the variants for the body component
+const bodyVariants = cva("components", {
+  variants: {
```
Comment (asanehisa) [inline src/components/puck/atoms/body.tsx:7]: so each of the variants i'd add a default case that is ""

File: src/components/puck/atoms/body.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    color: {
+      default: "",
+      primary: "text-primary",
+      secondary: "text-secondary",
+      accent: "text-accent",
+      text: "text-text",
+      background: "text-primary-background",
+    },
+    textTransform: {
+      none: "ve-normal-case",
```
Comment (asanehisa) [inline src/components/puck/atoms/body.tsx:29]: i think this could be "" too?

File: src/components/puck/atoms/body.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    VariantProps<typeof bodyVariants> {
+  textSize?: number;
+}
+
+const Body = React.forwardRef<HTMLParagraphElement, BodyProps>(
+  (
+    { className, textSize, fontWeight, color, textTransform, ...props },
+    ref
+  ) => {
+    const dynamicStyles = {
```
Comment (asanehisa) [inline src/components/puck/atoms/body.tsx:54]: oh sorry adjust this to my latest one that accounts for default

**Review by benlife5 — APPROVED** [review]

**Review by asanehisa — APPROVED** [review]

**Pull request merged**